src/subgraph/markets/index.ts

```diff
@@ -1,24 +1,37 @@
 import { request } from 'graphql-request';
 import { Chain } from '../../common/chain';
-import { getSubgraphEndpoint } from '../common';
+import { Market, getSubgraphEndpoint } from '../common';
 import { fetchAllMarketsDataQuery, fetchMarketByIdQuery } from './subgraphQueries';
 import { mapMarketsArrayToInterface, mapSingleMarketToInterface } from '../common/mapper';
 
-export const getAllMarketsFromSubgraph = async (chainId: Chain) => {
-  const subgraphEndpoint = getSubgraphEndpoint(chainId);
-  console.log('subgraphEndpoint', subgraphEndpoint);
-
-  const subgraphResponse = await request(subgraphEndpoint, fetchAllMarketsDataQuery);
-
-  return mapMarketsArrayToInterface(subgraphResponse);
+export const getAllMarketsFromSubgraph = async (chainId: Chain): Promise<Market[]> => {
+  try {
+    const subgraphEndpoint = getSubgraphEndpoint(chainId);
+    const subgraphResponse = await request(subgraphEndpoint, fetchAllMarketsDataQuery);
+    const markets = mapMarketsArrayToInterface(subgraphResponse);
+    if (markets) {
+      return markets;
+    }
+    throw new Error('Markets not found');
+  } catch (error) {
+    console.log('Markets not found');
+    throw error;
```

Comment:
We need to show 404 Http status code when this type of error occurs. Please create custom NotFoundError class and set status to 404 there.
refer - https://github.com/Parifi/parifi-microservices/blob/develop/src/error/resource-exists.error.ts